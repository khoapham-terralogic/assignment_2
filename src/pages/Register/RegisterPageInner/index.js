import React, { useEffect } from 'react';
import RegisterForm from '../RegisterForm';
import PropTypes from 'prop-types'
import { useHistory } from 'react-router-dom';
import { ClipSpinner } from '../../../components';

const RegisterPageInner = ({
    registerUser,
    isLoading,
    isAuth,
    msg
}) => {
    const history = useHistory()
    useEffect(() => {
        if (isAuth) history.push("/user")
    }, [msg, isAuth, history])
    return (
        <div className="login">
            {isLoading ? <div className="loadingBackground"><ClipSpinner size={40} color="#fff" /></div> : null}
            <div className="login-header">register your account</div>
            <RegisterForm registerUser={registerUser} />
        </div>
    );
}

RegisterPageInner.propTypes = {
    isAuth: PropTypes.bool.isRequired,
    isLoading: PropTypes.bool.isRequired,
    registerUser: PropTypes.func.isRequired,
    msg: PropTypes.string,
}


export default RegisterPageInner
