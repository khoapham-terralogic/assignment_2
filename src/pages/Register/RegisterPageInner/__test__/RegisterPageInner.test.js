import React from 'react'
import { shallow } from 'enzyme';
import RegisterPageInner from '..';

afterEach(() => {
    jest.resetAllMocks()
})

describe('Test <RegisterPageInner />', () => {
    const props = {
        registerUser: jest.fn(),
        isLoading: false,
        isAuth: false,
        msg: "",
    }
    const wrapper = shallow(<RegisterPageInner {...props} />)
    it('should render', () => {
        expect(wrapper).toMatchSnapshot()
    });
    it('should render loading component if isLoading is true', () => {
        expect(wrapper.find(".loadingBackground")).toHaveLength(0)
        wrapper.setProps({ isLoading: true })
        expect(wrapper.find(".loadingBackground")).toHaveLength(1)
        expect(wrapper.find(".loadingBackground")).toMatchSnapshot()
    });
    it('should redirect /user loading component if isAuth is true', () => {
        let useEffect
        const mockUseEffect = () => {
            useEffect.mockImplementationOnce(f => f());
        };
        beforeEach(() => {
            useEffect = jest.spyOn(React, "useEffect").mockImplementation(f => f());
            mockUseEffect();
            mockUseEffect();
            expect(wrapper.find(RegisterPageInner)).toHaveLength(1)
            wrapper.setProps({ isAuth: true })
            expect(wrapper.find(RegisterPageInner)).toHaveLength(0)
        });
    });
})
