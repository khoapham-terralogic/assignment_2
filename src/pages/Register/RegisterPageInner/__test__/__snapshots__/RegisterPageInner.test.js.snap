// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <RegisterPageInner /> should render 1`] = `
<div
  className="login"
>
  <div
    className="login-header"
  >
    register your account
  </div>
  <WithFormik(MyForm)
    registerUser={[MockFunction]}
  />
</div>
`;

exports[`Test <RegisterPageInner /> should render loading component if isLoading is true 1`] = `
<div
  className="loadingBackground"
>
  <ClipSpinner
    color="#fff"
    size={40}
  />
</div>
`;
